projects/vdk-plugins/vdk-audit/src/vdk/plugin/audit/audit_plugin.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import sys
+from typing import List
+
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import IPluginRegistry
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.plugin.audit.audit_config import add_definitions
+from vdk.plugin.audit.audit_config import AuditConfiguration
+
+
+class AuditPlugin:
+    @staticmethod
+    @hookimpl
+    def vdk_configure(config_builder: ConfigurationBuilder) -> None:
+        add_definitions(config_builder)
+
+    @hookimpl
+    def initialize_job(self, context: JobContext) -> None:
+        self._config = AuditConfiguration(context.core_context.configuration)
+
+        logging.getLogger(__name__).error("Start:")
+        logging.getLogger(__name__).error(not self._config.enabled())
+        if not self._config.enabled():
+            return
+
+        not_permitted_events_list = self._config.not_permitted_events_list().split(";")
+
+        def _audit(event, args):
+            if any(
+                event in not_permitted_event
+                for not_permitted_event in not_permitted_events_list
+            ):
+                logging.getLogger(__name__).warning(
+                    f'[Audit] Detected NOT permitted operation "{event}" with '
+                    f'arguments "{args}" '
+                )
+
+                if self._config.exit_on_not_permitted_event():
```

Comment:
I think this line looks a bit weird. 
If something is in the not permitted events you would think it should always terminate the job.

Reply:
Agreed, I put it just in case we want only to monitor the job.

Reply:
@tozka What is your opinion?

Reply:
I think it's a very good idea. It's a required feature for using the audit functionality in any pre-existing (brownfield) environment. I imagine an adoption would be 

1. Enable the audit hook with exit_on_forbidden_event == false 
2. Analyze how many jobs are violating the rules and which events
3. Fix the jobs or amend the forbidden events and back to 2 
4. Eventually enable the hook to terminate jobs when you know it will work safely for all jobs.

Also, it's useful when debugging/fixing a single job to be compliant. So that you can see all violations at once (when there is more than one) and not: fix the first one, re-run to see the second one, fix the second, re-run ... .